app/controllers/conversations.js

```diff
@@ -48,23 +48,31 @@ class ConversationsController extends BaseController {
         $or: [
           {
             type: "u",
-            owner_id: ObjectId(this.sessionRepository.getSessionUserId(ws)),
+            owner_id: ObjectId(currentUserId),
             opponent_id: conversationParams.opponent_id,
           },
           {
             type: "u",
             owner_id: ObjectId(conversationParams.opponent_id),
-            opponent_id: this.sessionRepository.getSessionUserId(ws),
+            opponent_id: currentUserId,
           },
         ],
       });
-      if (existingConversation)
+
+      if (existingConversation) {
+        const participant = new ConversationParticipant({
```

Comment:
There are 2 issues with this code:

1) what if both users are already in participants, will it create a new 3rd participant record?
2) it should be able to restore not only current user but also an opponent user, cause both users can delete a chat and both users can re-create it.